backend/utils/db.py

```diff
@@ -22,93 +24,131 @@ def get_db_connection(db_file_path:str="data/db.sqlite3"):
 
 def init_db(db_file_path: str = "data/db.sqlite3"):
     """Initialize the database with schema"""
-    os.makedirs(os.path.dirname(db_file_path), exist_ok=True)
-    with open('data/schema.sql') as f:
-        schema = f.read()
-    with get_db_connection() as conn:
-        conn.executescript(schema)
+    try:
+        os.makedirs(os.path.dirname(db_file_path), exist_ok=True)
+        with open('data/schema.sql') as f:
+            schema = f.read()
+        with get_db_connection() as conn:
+            conn.executescript(schema)
+    except sqlite3.Error as e:
+        raise DatabaseInitError(f"Error when initializing database: {str(e)}")
 
 
 def get_inventory() -> list[Snack]:
     """Returns all snacks in the database"""
-    with get_db_connection() as conn:
-        cursor = conn.cursor()
-        cursor.execute("SELECT * FROM snacks")
-        records = cursor.fetchall()
-        return [Snack(**record) for record in records]
+    try:
+        with get_db_connection() as conn:
+            cursor = conn.cursor()
+            cursor.execute("SELECT * FROM snacks")
+            records = cursor.fetchall()
+            return [Snack(**record) for record in records]
+    except sqlite3.Error as e:
+        raise DatabaseError(f"Database error when fetching snacks: {str(e)}")
 
 
 def get_snack(sku: str) -> Snack:
     """Returns a single snack by SKU"""
-    with get_db_connection() as conn:
-        cursor = conn.cursor()
-        cursor.execute("SELECT * FROM snacks WHERE sku = ?", (sku,))
-        record = cursor.fetchone()
-        return Snack(**record)
-
+    try:
+        with get_db_connection() as conn:
+            cursor = conn.cursor()
+            cursor.execute("SELECT * FROM snacks WHERE sku = ?", (sku,))
+            record = cursor.fetchone()
+            if record is None:
+                raise RecordNotFoundError(f"No snack found with SKU: {sku}")
+            return Snack(**record)
+    except sqlite3.Error as e:
+        if(isinstance(e,ConnectionError)):
+            raise ConnectionError(f"Error when connecting to database: {str(e)}")
+        raise DatabaseError(f"Database error when fetching snack {sku}: {str(e)}")
```

Comment:
I *think* you can do something like this which makes it more readable
```suggestion
except ConnectionError as e:
        raise ConnectionError(f"Error when connecting to database: {str(e)}")
except Exception as e:
        raise DatabaseError(f"Database error when fetching snack {sku}: {str(e)}")